### ISSUE TITLE
Add cloud-controller support

### ISSUE THREAD

**Issue by brandond** [rancher/k3k#40]
K3k should have a cloud controller integration:
* Disable the k3s built-in cloud provider, deploy k3k cloud controller (via injected deployment manifest) instead
* Inner cluster cloud controller communicates with outer controller using pod serviceaccount credentials (mounted from host).
  * Cluster pod list backs node lifecycle hooks (node exists, IP list, hostname, ID, etc)
  * LB controller creates LB services in outer cluster for LB services in inner cluster

**Comment by galal-hussein** [comment]
@brandond we need to keep in mind that the host cluster may not necessarily  be a k3s or rke2 server, ideally we want to provide support for all distros as a host cluster, so exposing services as LB services on the host will be a bit challenging

**Comment by brandond** [comment]
Yeah, and RKE2 doesn't come with a loadbalancer controller at all, so that wouldn't even work out of the box on RKE2.

I guess perhaps we should allow the user to decide what type of service they want created on the outer cluster? Could be useful to both set the default, and to override that default with an annotation. Maybe also annotation to set the traffic policy on the outer service?

Can you think of any other ways that someone might want to expose a k3k service on the outer cluster?

**Comment by rossella** [comment]
Revise this in the future. It's out of scope for now.
